android-components/components/feature/media/src/main/java/mozilla/components/feature/media/middleware/RecordingDevicesMiddleware.kt

```diff
@@ -157,7 +158,7 @@ class RecordingDevicesMiddleware(
         context.registerReceiverCompat(
             broadcastReceiver,
             IntentFilter(ACTION_RECORDING_DEVICES_NOTIFICATION_DISMISSED),
-            ContextCompat.RECEIVER_NOT_EXPORTED,
+            ContextCompat.RECEIVER_EXPORTED,
```

Comment:
Checking at line 137 to 140: 

```kotlin
        val intent =
            context.packageManager.getLaunchIntentForPackage(context.packageName)?.apply {
                flags = Intent.FLAG_ACTIVITY_NEW_TASK
} ?: throw IllegalStateException("Package has no launcher intent")

```

This looks safe as we are passing the `packageName`.

Reply:
Not a big deal right now, but don't those lines also mean that other applications can observe and receive updates when a user swipes away our notification? Shouldn't the intent be non-exported in that case (so that it matches this registerReceiverCompat as well)?

Reply:
It's the other way around, the receiver can receive broadcasts from other apps not just our own, when the exported flag is used.

To answer the other questions, the intent received is the one with `ACTION_RECORDING_DEVICES_NOTIFICATION_DISMISSED`, that does not have a package set. I tried to set the package to make it an explicit intent, but it is still not received. Without investigating further I would say that it is because the system actually sends the intent set for the delete intent, not our app.

That means that the receiver has to receive broadcasts from anywhere, thus the reason for the filtering in `onReceive`. 

Side note, the documentation says that for broadcasts received exclusively from the system you don't have to specify a flag. From my testing and several bugs filed on the Google issue tracker, it seems that causes crashes even if the docs say they shouldn't.